### ISSUE TITLE
Encoded file is longer than original 

### ISSUE THREAD

**Issue by danielbanfield** [Garciat/libvorbis.js#6]
Hi,
Thanks for the great lib.
I've read that lossless encoding can add time to a file but wondering if there is something else going on. I took code pretty much straight from the example. I've tried a range of input files (although all from the same wav file pack) and get changes like
Input: 5.33 secs
Output: 5.97 secs

Just want to confirm if that kind of additional length is expected - i need precise length of original as use case is music looping.
Thanks
Dan

**Comment by Garciat** [comment]
I tried encoding [this](http://freewavesamples.com/alesis-fusion-acoustic-bass-c2) WAV file with `examples/recode.html`, and, yes, the resulting OGG file is longer:

```
$ soxi -D Alesis-Fusion-Acoustic-Bass-C2.wav 
2.897551
$ soxi -D Alesis-Fusion-Acoustic-Bass-C2.ogg
3.715193
```

I have no particular knowledge of either Vorbis or audio encoding in general to intuit a fix for this, sorry. If you do, let me know, and we can work something out.

My test files:
https://dl.dropboxusercontent.com/u/1427833/Alesis-Fusion-Acoustic-Bass-C2.wav
https://dl.dropboxusercontent.com/u/1427833/Alesis-Fusion-Acoustic-Bass-C2.ogg

**Comment by danielbanfield** [comment]
Ok, no problem. I too am no expert on audio encoding :) I've done some investigating - no fix yet but promising characteristics:
- I took a 4 second loop and ogg encoded it using FFMPEG. The output was exactly 4 seconds! This reassures me that we aren't dealing with some subtle oddity of the encoding process 
- I took a look at both WAV and OGG files in a waveform inspector. It looks like the additional length is purely from a fraction of silence added on the end. This suggests there may just be something up with the chunking part of the code taken from the example.

I'll have a closer look at the chunking code, will post anything I find.

**Comment by danielbanfield** [comment]
Solved! :D

There is a subtle bug in the example.

Lines 98 and 100

```
var actualSize = Math.min(chunkSize, samples - n);
var chunkBuffer = ctx.createBuffer(channels, chunkSize, ctx.sampleRate);
```
- chunkBuffer is created using chunkSize. It should be created using actualSize which uses a min to get the size of the last chunk, otherwise the encode method on line 66 receives an audioBuffer that is chunkSize, making the last chunk get encoded with a samples count of chunkSize (hence the additional silence)
- I've no idea why, but with experimentation the actualSize for the last chunk needs to be computed 1 less
  `var actualSize = Math.min(chunkSize, samples - n - 1);`

Relieved I didn't have to go diving into vorbis code :)

**Comment by Garciat** [comment]
I'll gladly accept a PR with the fix (:

Let me know otherwise, and I'll make the fix, no prob.

Thanks !

**Comment by Garciat** [comment]
Fixed by #7
